File: 118.Pascal'sTriangle.cpp
class Solution
{
public:
    vector<vector<int>> generate(int numRows)
    {
        if (numRows == 0)
        {
            return {};
        }
        vector<vector<int>> result = {{1}};
        for (int i = 0; i < numRows - 1; ++i)
        {
            vector<int> temp;
            temp.push_back(result.back()[0]);
            for (int j = 1; j < result.back().size(); ++j)
            {
                temp.push_back(result.back()[j - 1] + result.back()[j]);
            }
            temp.push_back(result.back().back());
            result.push_back(temp);
        }
        return result;
    }
};

File: 118.Pascal'sTriangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        if numRows==0:
            return []
        result = [[1]]
        for i in range(numRows-1):
            temp = []
            temp.append(result[-1][0])
            for j in range(1,len(result[-1])):
                temp.append(result[-1][j-1]+result[-1][j])
            temp.append(result[-1][-1])
            result.append(temp)
        return result